### PR TITLE
feat(rust): tcp inlet creation will always optional validate unless `--no-connection-wait` is used`

### PR DESCRIPTION
**Pull request by davide-baldo**
- default create inlet behavior is to always validate but to keep trying anyway
- `--no-validation` is now `--no-connection-wait`

Screenshot_20240215_133906

Screenshot_20240215_133035

### CODE REVIEW

File: implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs [hunk 1]
```diff
@@ -203,7 +203,21 @@ impl CreateCommand {
 
         opts.terminal
             .stdout()
-            .plain(
+            .plain(if let Some(last_failure) = inlet.last_failure.as_deref() {
+                fmt_err!(
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs]: Since this error it's not really a failure that forces the user to redo any action or do anything about it, I would use `fmt_warn`

File: implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs [hunk 2]
```diff
@@ -203,7 +203,21 @@ impl CreateCommand {
...
+                    &cmd.from
+                        .to_string()
+                        .color(OckamColor::PrimaryResource.color()),
+                    &node.node_name().color(OckamColor::PrimaryResource.color()),
+                    &cmd.to
+                        .to_string()
+                        .color(OckamColor::PrimaryResource.color())
+                ) + &fmt_log!(
+                    "Error: {}\n",
+                    last_failure.color(OckamColor::Failure.color())
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs]: This message can be misleading and confusing. We could log it (you already [added](https://github.com/build-trust/ockam/pull/7587/files#diff-5b3f34a98f0adf81c262908c1b6ee444f6e61e4242075b18860c9a57ea986e3fR409) this right?) but not write it to stdout.

File: implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs [hunk 3]
```diff
@@ -215,8 +229,21 @@ impl CreateCommand {
                     &cmd.to
                         .to_string()
                         .color(OckamColor::PrimaryResource.color())
-                ),
-            )
+                )
+            } else {
+                fmt_ok!(
+                    "TCP Inlet {} on node {} will automatically connect\n",
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs]: To keep the messages consistent, we should use `inlet` (in lowercase). For example: `The inlet {} on node {} will automatically connect`.

I would also rephrase this to `The inlet {} on node {} will automatically connect when the outlet becomes available`.

**Comment by adrianbenavides** [comment]
Also, could you add a simple bats test as an example of this? Something along these lines:

```
ockam node create n2
ockam tcp-inlet create --at /node/n2 --from 127.0.0.1:6000 --to /ip4/127.0.0.1/tcp/4000/service/outlet
ockam node create n1 -t 127.0.0.1:4000
ockam tcp-outlet create --at /node/n1 --to 127.0.0.1:5000
sleep 5
curl 127.0.0.1:6000
```

File: implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs (hunk 2 above)
Comment (davide-baldo) [reply implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs]: Maybe i can show the "last failure" when listing/showing the inlets

Comment (davide-baldo) [reply implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs]: After some experimentation, I'm completely removing `last_failure`. The errors are too rough to for the stdout.

**Comment by davide-baldo** [comment]
@adrianbenavides 
I've pushed an updated version and updated the screenshots as well

File: implementations/rust/ockam/ockam_command/tests/bats/portals.bats [hunk 4]
```diff
@@ -228,3 +228,17 @@ teardown() {
...
+
+  run_success "$OCKAM" node create n1
+  run_success "$OCKAM" node create n2
+
+  run_success "$OCKAM" tcp-inlet create --at /node/n2 --from "127.0.0.1:$port" --to /node/n1/service/outlet
+  run_success "$OCKAM" tcp-outlet create --at /node/n1 --to 127.0.0.1:5000
+
+  sleep 5
+
+  run_success curl --fail --head --max-time 20 "127.0.0.1:$port"
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/tests/bats/portals.bats]: I think the n1 must be created after the inlet, otherwise it will connect right away.

File: implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs [hunk 5]
```diff
@@ -215,8 +214,32 @@ impl CreateCommand {
...
+                    &cmd.from
+                        .to_string()
+                        .color(OckamColor::PrimaryResource.color()),
+                    &node.node_name().color(OckamColor::PrimaryResource.color())
+                ) + &fmt_log!(
+                    "to the outlet at {}",
+                    &cmd.to
+                        .to_string()
+                        .color(OckamColor::PrimaryResource.color())
+                )
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs]: I just realized from checking the screenshots that this produces a weird message due to the second line (`"to the outlet at {}",`).

Can we change it to the following?

```
fmt_ok!(
    "The inlet {} on node {} will automatically connect when the outlet at {} is available\n",
    &cmd.from
        .to_string()
        .color(OckamColor::PrimaryResource.color()),
    &node.node_name().color(OckamColor::PrimaryResource.color()),
    &cmd.to
        .to_string()
        .color(OckamColor::PrimaryResource.color())
)
```

**Review by adrianbenavides — APPROVED** [review]
🚀
